libnmstate/error.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+try:
+    PermissionError
+except NameError:
+    # Python 2 does not have PermissionError.
+    class PermissionError(Exception):
+        pass
+
+
+class NmstateError(Exception):
+    """
+    The base exception of libnmstate.
+    """
+    def __init__(self, msg):
+        super(NmstateError, self).__init__(msg)
+        self._msg = msg
+
+    @property
+    def msg(self):
+        """
+        Error message of specified NmstateError exception.
+        """
+        return self._msg
+
+    def __str__(self):
+        return self._msg
+
+
+class NmstateExternalRequirementError(NmstateError):
```

Comment:
How about naming this `NmstateDependencyMissingError`?

Reply:
Dependency seems like a packaging issue.

Reply:
The description says that it is about missing external tools or started services which happen when packaging is broken... Maybe just `NmstateDependencyError`?

Reply:
Changed to `NmstateDependencyError`.